ansible/plugins/connection/switch.py

```diff
@@ -242,14 +247,14 @@ def exec_command(self, *args, **kwargs):
             i = client.expect(['\(y\/n\)\??\s*\[n\]', 'Proceed with reload\? \[confirm\]', 'System configuration has been modified. Save\? \[yes\/no\/cancel\/diff\]:'])
             if i == 2:
                 # EOS behavior
-                stdout += client.before
+                stdout += self._remove_unprintable(client.before)
                 client.sendline('n')
                 i = client.expect('Proceed with reload\? \[confirm\]')
-            stdout += client.before
+            stdout += self._remove_unprintable(client.before)
             client.sendline('y')
             # The system is going down for reboot NOW: EOS
             i = client.expect(['>', '#', 'The system is going down for reboot NOW', pexpect.TIMEOUT, pexpect.EOF])
-            stdout += client.before
+            stdout += self._remove_unprintable(client.before)
```

Comment:
Similar here, keep the original `stdout += client.before`. When all the `client.before` is collected, use below code to change its encoding to ASCII.
```
stdout = stdout.encode('ascii', 'backslashreplace')
```

Reply:
Basically it is failing on command "stdout += client.before", because it can't perform concatination as 'client.before' contains unprintable characters (> 128 in ASCII table).

Reply:
I think there is no such issue if initializing variable `stdout` as unicode string, for exmple:
```
stdout=u''
```

Reply:
Of course that 'stdout' variable can be casted to unicode and there will be no concatenation issue. However in Python 2 the default encoding is ASCII, so I suggest to remove unprintable symbols to avoid farther possible issues or inconveniences with variable processing. Or lets add someone else to vote :)

Reply:
While concatenating the output, we can keep using unicode. Before printing the string as ASCII encoding, there are two options to convert the string:
1. Remove unprintable characters.
2. Use the python built-in to convert the encoding to ASCII.
IMO, option 2 looks better.

Reply:
If we use option 2, there is still chance that we can convert the ASCII string back to unicode without information loss. But for this scenario, it is unlikely we need to convert it back to unicode again. Removing symbols > 128 seems harmless. Let's keep it as it is.